cpp/include/cudf/utilities/error.hpp

```diff
@@ -237,7 +237,7 @@ inline void throw_cuda_error(cudaError_t error, char const* file, unsigned int l
   // Calls cudaGetLastError to clear the error status. It is nearly certain that a fatal error
   // occurred if it still returns the same error after a cleanup.
   cudaGetLastError();
-  auto const last = cudaFree(0);
+  auto const last = cudaFree(nullptr);
```

Comment:
Could this be detected using compiler warnings?

Reply:
IIRC this can be detected with host compilers but not nvcc. I'd have to double-check that, though.

Reply:
OK, yes, this can be enforced with `-Wzero-as-null-pointer-constant`. However, that leads to warnings from `cuda_runtime_api.h`. I'm not sure if there is a good way to avoid that.